system/camerad/cameras/camera_qcom2.cc

```diff
@@ -927,41 +927,50 @@ void CameraState::set_camera_exposure(float grey_frac) {
   sensors_i2c(exp_reg_array.data(), exp_reg_array.size(), CAM_SENSOR_PACKET_OPCODE_SENSOR_CONFIG, ci->data_word);
 }
 
-static void process_driver_camera(MultiCameraState *s, CameraState *c, int cnt) {
-  c->set_camera_exposure(set_exposure_target(&c->buf, c->ae_xywh, 2, 4));
-
-  MessageBuilder msg;
-  auto framed = msg.initEvent().initDriverCameraState();
-  fill_frame_data(framed, c->buf.cur_frame_data, c);
-
-  c->ci->processRegisters(c, framed);
-  s->pm->send("driverCameraState", msg);
-}
+void CameraState::run(PubMaster *pm, const char *pub_name) {
+  util::set_thread_name(pub_name);
+
+  // Get the member function pointer for initializing the camera state based on the stream type
+  auto initCameraState = [type = buf.stream_type]() {
+    if (type == VISION_STREAM_DRIVER) return &cereal::Event::Builder::initDriverCameraState;
+    if (type == VISION_STREAM_ROAD) return &cereal::Event::Builder::initRoadCameraState;
+    return &cereal::Event::Builder::initWideRoadCameraState; }();
```

Comment:
let's avoid any if's like this. patterns like we use for configuring the encoders is better

Reply:
This might make this PR too large. There's still a lot of work to be done to clean up the camerad code, which will require a series of PRs. As a start, I have submitted a PR to add a CameraConfig struct for initializing CameraState in the constructor. https://github.com/commaai/openpilot/pull/33034